api/src/oscc.c

```diff
@@ -1092,3 +1092,84 @@ oscc_result_t get_device_name( char * string, char * const name )
 
     return result;
 }
+
+oscc_result_t get_wheel_speed_right_rear(
+    struct can_frame const * const frame,
+    double * wheel_speed_right_rear)
+{
+    if((frame == NULL) || (wheel_speed_right_rear == NULL)) { return OSCC_ERROR; }
+    if(frame->can_id != KIA_SOUL_OBD_WHEEL_SPEED_CAN_ID) { return OSCC_ERROR; }
+    uint16_t raw = ((frame->data[7] & 0x0F) << 8) | frame->data[6];
+    // 10^-1 precision, raw / 32.0
+    *wheel_speed_right_rear = (double)((int)((double)raw / 3.2) / 10.0);
+    return OSCC_OK;
+}
+
+
+oscc_result_t get_wheel_speed_left_rear(
+    struct can_frame const * const frame,
+    double * wheel_speed_left_rear)
+{
+    if((frame == NULL) || (wheel_speed_left_rear == NULL)) { return OSCC_ERROR; }
+    if(frame->can_id != KIA_SOUL_OBD_WHEEL_SPEED_CAN_ID) { return OSCC_ERROR; }
+    uint16_t raw = ((frame->data[5] & 0x0F) << 8) | frame->data[4];
+    // 10^-1 precision, raw / 32.0
+    *wheel_speed_left_rear = (double)((int)((double)raw / 3.2) / 10.0);
+    return OSCC_OK;
+}
+
+
+oscc_result_t get_wheel_speed_right_front(
+    struct can_frame const * const frame,
+    double * wheel_speed_right_front)
+{
+    if((frame == NULL) || (wheel_speed_right_front == NULL)) { return OSCC_ERROR; }
+    if(frame->can_id != KIA_SOUL_OBD_WHEEL_SPEED_CAN_ID) { return OSCC_ERROR; }
+    uint16_t raw = ((frame->data[3] & 0x0F) << 8) | frame->data[2];
+    // 10^-1 precision, raw / 32.0
+    *wheel_speed_right_front = (double)((int)((double)raw / 3.2) / 10.0);
+    return OSCC_OK;
+}
+
+
+oscc_result_t get_wheel_speed_left_front(
+    struct can_frame const * const frame,
+    double * wheel_speed_left_front)
+{
+    if((frame == NULL) || (wheel_speed_left_front == NULL)) { return OSCC_ERROR; }
+    if(frame->can_id != KIA_SOUL_OBD_WHEEL_SPEED_CAN_ID) { return OSCC_ERROR; }
+    uint16_t raw = ((frame->data[1] & 0x0F) << 8) | frame->data[0];
+    // 10^-1 precision, raw / 32.0
+    *wheel_speed_left_front = (double)((int)((double)raw / 3.2) / 10.0);
+    return OSCC_OK;
+}
+
+
+oscc_result_t get_steering_wheel_angle(
+    struct can_frame const * const frame,
+    double * steering_wheel_angle)
+{
+    if((frame == NULL) || (steering_wheel_angle == NULL)) { return OSCC_ERROR; }
+    if(frame->can_id != KIA_SOUL_OBD_STEERING_WHEEL_ANGLE_CAN_ID) { return OSCC_ERROR; }
+    int16_t raw = (frame->data[1] << 8) | frame->data[0];
+    *steering_wheel_angle = -((double)raw / 10.0);
+    return OSCC_OK;
+}
+
+
+oscc_result_t get_brake_pressure(
+    struct can_frame const * const frame,
+    double * brake_pressure)
+{
+    if((frame == NULL) || (brake_pressure == NULL)) { return OSCC_ERROR; }
+    if(frame->can_id != KIA_SOUL_OBD_BRAKE_PRESSURE_CAN_ID) { return OSCC_ERROR; }
+#ifdef KIA_NIRO
```

Comment:
This seems like an abstraction leaked, is it a good use of time to extract this variance to individual vehicle files?

Reply:
This was a design decision with @rebpdx for minimal touch. Moving to individual header files represents a few difficulties. 

- First, except for this instance for the Kia Niro, it's all duplicated code.
- Second, there would potentially have to be a larger refactor required so that individual vehicle headers had access to the `oscc_result_t` type. (`oscc.h` includes the vehicle headers that themselves need to include `oscc.h` for the `oscc_result_t` type. The critical circular dependency is managed by the header guards but that means the data type isn't defined before the type is used, woo!)